iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachNewLineOnLine.java

```diff
@@ -0,0 +1,331 @@
+    }
+
+    @Override
+    public void apply(Network network) {
+        // Create the fictitious voltage Level at the attachment point
+        VoltageLevel fictitiousVl;
+        if (createFictSubstation) {
+            fictitiousVl = network.newSubstation()
+                    .setId(fictitiousSubstationId)
+                    .setName(fictitiousSubstationName)
+                    .setFictitious(true)
+                    .add()
+                    .newVoltageLevel()
+                    .setId(fictitiousVlId)
+                    .setName(fictitiousVlName)
+                    .setFictitious(true)
+                    .setNominalV(line.getTerminal1().getVoltageLevel().getNominalV())
+                    .setTopologyKind(TopologyKind.NODE_BREAKER)
+                    .add();
+        } else {
+            fictitiousVl = network.newVoltageLevel()
+                    .setId(fictitiousVlId)
+                    .setName(fictitiousVlName)
+                    .setFictitious(true)
+                    .setNominalV(line.getTerminal1().getVoltageLevel().getNominalV())
+                    .setTopologyKind(TopologyKind.NODE_BREAKER).add();
+        }
+
+        // Create the two lines replacing the existing line
+        LineAdder adder1 = createLineAdder(percent, line1Id, line1Name, line.getTerminal1().getVoltageLevel().getId(), fictitiousVlId, network, line);
+        LineAdder adder2 = createLineAdder(100 - percent, line2Id, line2Name, fictitiousVlId, line.getTerminal2().getVoltageLevel().getId(), network, line);
+        attachLine(line.getTerminal1(), adder1, (bus, adder) -> adder.setConnectableBus1(bus.getId()), (bus, adder) -> adder.setBus1(bus.getId()), (node, adder) -> adder.setNode1(node));
+        attachLine(line.getTerminal2(), adder2, (bus, adder) -> adder.setConnectableBus2(bus.getId()), (bus, adder) -> adder.setBus2(bus.getId()), (node, adder) -> adder.setNode2(node));
+        Line line1 = adder1.setNode2(0).add();
+        Line line2 = adder2.setNode1(2).add();
+        addLoadingLimits(line1, line, Branch.Side.ONE);
+        addLoadingLimits(line2, line, Branch.Side.TWO);
+
+        // Create the topology inside the fictitious voltage level
+        fictitiousVl.getNodeBreakerView()
+                .newInternalConnection()
+                .setNode1(0)
+                .setNode2(1)
+                .add();
+        fictitiousVl.getNodeBreakerView()
+                .newInternalConnection()
+                .setNode1(1)
+                .setNode2(2)
+                .add();
+        fictitiousVl.getNodeBreakerView()
+                .newInternalConnection()
+                .setNode1(1)
+                .setNode2(3)
+                .add();
+
+        // Set the end points of the new line
+        lineAdder.setNode1(3).setVoltageLevel1(fictitiousVlId).setVoltageLevel2(voltageLevelId);
+
+        // Create topology in the existing voltage level
+        VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelId);
```

Comment:
We should check and throw an exception in case of voltage level not found. I suspect a NPE here in that case.